From 388ede48f00c3b15e80708d2da5bbcaf8ada28b9 Mon Sep 17 00:00:00 2001
From: Akash Patel <17132214+acxz@users.noreply.github.com>
Date: Sun, 31 Jul 2022 21:46:27 -0400
Subject: [PATCH] set big endian default for ieee_double_extract

when HAVE_DOUBLE_* is not defined, ieee_double_extract fails to be defined and mpir compilation fails
set a default value of big endian in such case
---
 gmp-impl.h | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/gmp-impl.h b/gmp-impl.h
index a7c6e922a..89a4f2d47 100644
--- a/gmp-impl.h
+++ b/gmp-impl.h
@@ -3384,9 +3384,8 @@ union ieee_double_extract
     } s;
   double d;
 };
-#endif
 
-#if HAVE_DOUBLE_IEEE_LITTLE_ENDIAN
+#elif HAVE_DOUBLE_IEEE_LITTLE_ENDIAN
 #define _GMP_IEEE_FLOATS 1
 union ieee_double_extract
 {
@@ -3399,9 +3398,8 @@ union ieee_double_extract
     } s;
   double d;
 };
-#endif
 
-#if HAVE_DOUBLE_IEEE_BIG_ENDIAN
+#else
 #define _GMP_IEEE_FLOATS 1
 union ieee_double_extract
 {
